### PR TITLE
Deploy Vector agent for Reaper metrics scraping

### LINKED ISSUE #789: Deploy Vector agent for Reaper metrics scraping
Following up on #775, we'd need to deploy Vector as a sidecar in the Reaper container in order to scrape its metrics and forward them based on a user provided configuration.

The api changes will be done in #775 in the telemetry spec.

### PR DESCRIPTION
**Pull request by emerkle826**
**What this PR does**:
Adds ability to configure Vector as a sidecar for Reaper
**Which issue(s) this PR fixes**:
Fixes #789 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: test/e2e/reaper_test.go [hunk 1]
```diff
@@ -35,6 +39,36 @@ func createSingleReaper(t *testing.T, ctx context.Context, namespace string, f *
...
+	t.Logf("check that if Reaper Vector is enabled, the agent and configmap are re-created")
+	err = f.Client.Get(ctx, kcKey, kc)
+	require.NoError(err, "failed to get K8ssandraCluster in namespace %s", namespace)
+	reaperVectorPatch = client.MergeFromWithOptions(kc.DeepCopy(), client.MergeFromWithOptimisticLock{})
+	kc.Spec.Reaper.Telemetry.Vector.Enabled = pointer.Bool(true)
+	err = f.Client.Patch(ctx, kc, reaperVectorPatch)
+	require.NoError(err, "failed to patch K8ssandraCluster in namespace %s", namespace)
+	checkReaperReady(t, f, ctx, reaperKey)
+	checkReaperK8cStatusReady(t, f, ctx, kcKey, dcKey)
+	checkVectorAgentPresence(t, ctx, f, reaperKey, getPodTemplateSpecForReaper, reaper.VectorContainerName)
```
Comment (olim7t) [inline test/e2e/reaper_test.go:69]: I know it's not part of this PR, but looking at the implementation of the function:
```go
func checkVectorAgentPresence(t *testing.T, ctx context.Context, f *framework.E2eFramework, podKey framework.ClusterKey, specFunction func(t *testing.T, ctx context.Context, f *framework.E2eFramework, dcKey framework.ClusterKey) *corev1.PodTemplateSpec, vectorContainerName string) {
	t.Logf("check that %s contains Vector Agent %s", podKey.Name, vectorContainerName)
	podTempSpec := specFunction(t, ctx, f, podKey)
	_, containerFound := cassandra.FindContainer(podTempSpec, vectorContainerName)
	require.True(t, containerFound, "cannot find Vector agent container in pod template spec")
}
```
It doesn't look like there's anything specific to Vector, it only checks the container by name. So maybe we shouldn't have "Vector" in the function name.

**Review by olim7t — APPROVED** [review]
A few remarks but none of them blocking.

File: controllers/reaper/vector.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	reaperpkg "github.com/k8ssandra/k8ssandra-operator/pkg/reaper"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/telemetry"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/vector"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *ReaperReconciler) reconcileVectorConfigMap(
```
[the remark this answers is not in the document]
Comment (emerkle826) [reply controllers/reaper/vector.go:22]: I just created a ticket for it:
https://github.com/k8ssandra/k8ssandra-operator/issues/843

File: test/e2e/reaper_test.go (hunk 1 above)
Comment (emerkle826) [reply test/e2e/reaper_test.go:69]: I suppose we can. I wrote this one for the Stargate Vector work, along with the 2 functions that retrieve the PodTemplateSpec from the [CassDc](https://github.com/k8ssandra/k8ssandra-operator/blob/main/test/e2e/suite_test.go#L2049-L2054) and the Stargate [deployment](https://github.com/k8ssandra/k8ssandra-operator/blob/main/test/e2e/suite_test.go#L2056-L2061). But I guess it doesn't matter if the container it's looking for is a Vector container or something else.

**Pull request merged**